File: 2025/1/1/1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

const (
	filename = "input.txt"
)

func loadData() []string {
	data := make([]string, 0)
	file, err := os.Open(filename)
	if err != nil {
		panic(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		data = append(data, scanner.Text())
	}

	if err := scanner.Err(); err != nil {
		panic(err)
	}

	return data
}

func normalize(dial int) int {
	if dial > 99 {
		dial -= 100
	}

	if dial < 0 {
		dial += 100
	}

	if dial >= 0 && dial < 100 {
		return dial
	} else {
		return normalize(dial)
	}
}

func turn(dial int, direction string, value int) int {
	if direction == "L" {
		dial -= value
	} else {
		dial += value
	}

	dial = normalize(dial)

	return dial
}

func main() {
	data := loadData()
	dial := 50
	counter := 0
	for _, v := range data {
		direction := string(v[0])
		number, err := strconv.Atoi(v[1:])
		if err != nil {
			panic(err)
		}
		dial = turn(dial, direction, number)
		if dial == 0 {
			counter++
		}
	}

	fmt.Println(counter)
}

File: 2019/2/1.py
FILENAME = "input.txt"

def load_data():
    with open(FILENAME) as file:
        return file.read().split(",")


if __name__ == "__main__":
    data = load_data()
    data = [int(x) for x in data]
    data[1] = 12
    data[2] = 2
    for i in range(0, len(data), 4):
        v1 = data[i]
        if v1 == 99:
            break
        v2 = data[data[i+1]]
        v3 = data[data[i+2]]
        v4 = data[i+3]
        if v1 == 1:
            data[v4] = v2 + v3
        elif v1 == 2:
            data[v4] = v2 * v3

    print(data[0])
